### PR TITLE
Implement Forge's ModelLoader

### PR DESCRIPTION
**Pull request by rikka0w0**

### CODE REVIEW

File: patchwork-model-loader/src/main/java/net/minecraftforge/client/model/ModelLoaderRegistry2.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+import net.minecraft.client.render.model.json.ModelTransformation;
+import net.minecraft.client.render.model.json.Transformation;
+import net.minecraft.resource.ReloadableResourceManager;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.JsonHelper;
+
+import net.patchworkmc.impl.modelloader.PatchworkJsonUnbakedModel;
+
+public class ModelLoaderRegistry2 {
+	private static final Logger LOGGER = LogManager.getLogger();
```
Comment (TheGlitch76) [inline patchwork-model-loader/src/main/java/net/minecraftforge/client/model/ModelLoaderRegistry2.java:55]: If Forge's model loader registery is so good...

Comment (rikka0w0) [reply patchwork-model-loader/src/main/java/net/minecraftforge/client/model/ModelLoaderRegistry2.java:55]: Forge 1.14.4 has 2 ModelLoaders, one is from 1.13, the other is for 1.15 and future. I will only implement the latter one. ModelLoaderRegistry2 will be renamed to ModelLoaderRegistry in 1.15.

Comment (florensie) [reply patchwork-model-loader/src/main/java/net/minecraftforge/client/model/ModelLoaderRegistry2.java:55]: ModelLoaderRegistry 2: Electric Boogaloo

**Comment by rikka0w0** [comment]
I think we should make IModelData a separate module.

**Comment by rikka0w0** [comment]
The IModelData stuff and IForgeBakedModel will be in #189. Otherwise the model-loader package will be too big. I will continue this once #189 is merged.

**Comment by rikka0w0** [comment]
Since there are a lot of incompatible changes between 1.14.4 and 1.15/1.16 ModelLoader, I decided to have a fresh start once we move to 1.15 or 1.16.

**Pull request closed without merging**